src/utils/nlpUtils/tagKeywordListReader.py

```diff
@@ -0,0 +1,113 @@
+import pandas as pd
+import numpy as np
+import itertools
+
+'''
+  These methods are designed to process the file tag_keywords_lists.xlsx
+  This file contains several keywords grouped into classes (e.g., material, components, etc)
+  For each class, a tag is defined: during the NLP analysis, when these keywords are identified,
+  this tag will be associated to such word.
+  These methods are designed to run in sequence just once at the beginning of the NLP
+  pipeline), e.g.:
+    tagsDict = keyWordListGenerator('../../nlp/data/tag_keywords_lists.xlsx')
+    unitDict = extractUnits('../../nlp/data/tag_keywords_lists.xlsx')
+    tagsDictCleaned, acronymsDict = cleanTagDict(tagsDict)
+'''
+
+def keyWordListGenerator(fileName):
+  '''
+  Method designed to read the file and generate a dictionary which contains, for each tag,
+  the set of keywords that should be associate to such tag.
+  '''
+  # TODO: ==> check if lower, and lemma (e.g. for plural)!!!
```

Comment:
I guess this will be in the pattern generation.

Reply:
let's meet to discuss it

Reply:
resolve it in future development